aiohttp/web_urldispatcher.py

```diff
@@ -987,11 +987,11 @@ async def resolve(self, request: Request) -> AbstractMatchInfo:
                 return match_dict
             else:
                 allowed_methods |= allowed
+
+        if allowed_methods:
+            return MatchInfoError(HTTPMethodNotAllowed(method, allowed_methods))
         else:
```

Comment:
I dropped the else here, as it is redundant, and makes it a little more difficult to see that function will always return.